nexus/mgs-updates/src/rot_updater.rs

```diff
@@ -203,23 +210,218 @@ impl SpComponentUpdateHelper for ReconfiguratorRotUpdater {
     /// Checks if the component is already updated or ready for update
     fn precheck<'a>(
         &'a self,
-        _log: &'a slog::Logger,
-        _mgs_clients: &'a mut MgsClients,
-        _update: &'a PendingMgsUpdate,
+        log: &'a slog::Logger,
+        mgs_clients: &'a mut MgsClients,
+        update: &'a PendingMgsUpdate,
     ) -> BoxFuture<'a, Result<PrecheckStatus, PrecheckError>> {
-        // TODO-K: fill in the precheck
-        todo!()
+        async move {
+            // Verify that the device is the one we think it is.
```

Comment:
It would be really really nice to update/reset the inactive slot as well to make sure both slots have the same version for correctness. But I'll leave that for a follow up PR

Reply:
Maybe I'm misunderstanding this note, but I don't think we want to update the inactive slot too at this level, right? That would be a decision the planner should make after this update completes.

Reply:
Ah yeah, no, sorry! I had a random thought and just plonked it there so I wouldn't forget. I should have left that as a separate comment and not on that line 😅